include/inviwo/core/datastructures/geometry/decoratedmesh.h

```diff
@@ -92,37 +91,92 @@ struct BufferTrait {
     type t;
     operator type() const { return t; }
 #endif
+
+    BufferTrait() {}
+    virtual ~BufferTrait() = default;
+
+    template <typename Mesh>
```

Comment:
No need for the template here..

Reply:
For some reason it did not work without it on my machine (MSVC 17).

Reply:
It should, what error do you get?

Reply:
No error, it compiles fine. Problem is that neither of the constructors are called, the buffer_ pointer is  empty.

Reply:
Try casting *this to a Mesh

Reply:
When you call the constructor

Reply:
that worked